packages/jaeger-ui/src/utils/TreeNode.js

```diff
@@ -83,7 +83,18 @@ export default class TreeNode {
   }
 
   walk(fn, depth = 0) {
-    TreeNode.iterFunction(fn, depth)(this);
-    this.children.forEach(child => child.walk(fn, depth + 1));
+    const nodeStack = [];
+    let actualDepth = depth;
+    nodeStack.push({ node: this, depth: actualDepth });
+    const pushChildren = n => nodeStack.push({ node: n, depth: actualDepth });
+    while (nodeStack.length) {
+      const { node, nodeDepth } = nodeStack.pop();
```

Comment:
I think this is destructing the property name:

```js
const { node, depth: nodeDepth } = nodeStack.pop();
```

Reply:
I like it :)